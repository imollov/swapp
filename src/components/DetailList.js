import React from 'react'
import PropTypes from 'prop-types'

import { Box, Text } from 'rebass/styled-components'

const DetailList = ({ data }) =>
  data.map(({ field, value }) => (
    <Box pb={1} key={field}>
      <Text variant="body" color="detailLabel" as="a">{`${field}: `}</Text>
      <Text variant="body" color="detailValue" as="a">
        {value}
      </Text>
    </Box>
  ))

DetailList.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      field: PropTypes.string,
      value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    }).isRequired,
  ).isRequired,
}

export default DetailList
