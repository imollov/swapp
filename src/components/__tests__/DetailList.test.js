import React from 'react'
import { shallow } from 'enzyme'

import DetailList from '../DetailList'

describe('DetailList Component', () => {
  it('should match snapshot', () => {
    const details = [
      { field: 'foo', value: 'bar' },
      { field: 'foo', value: 'bar' },
    ]
    const wrapper = shallow(<DetailList data={details} />)

    expect(wrapper).toMatchSnapshot()
  })
})
