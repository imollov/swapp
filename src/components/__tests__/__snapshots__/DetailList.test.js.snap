// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailList Component should match snapshot 1`] = `
Array [
  <styled.div
    key="foo"
    pb={1}
  >
    <ForwardRef
      as="a"
      color="detailLabel"
      variant="body"
    >
      foo: 
    </ForwardRef>
    <ForwardRef
      as="a"
      color="detailValue"
      variant="body"
    >
      bar
    </ForwardRef>
  </styled.div>,
  <styled.div
    key="foo"
    pb={1}
  >
    <ForwardRef
      as="a"
      color="detailLabel"
      variant="body"
    >
      foo: 
    </ForwardRef>
    <ForwardRef
      as="a"
      color="detailValue"
      variant="body"
    >
      bar
    </ForwardRef>
  </styled.div>,
]
`;
